src/providers/DevicesProvider/AudioInputProvider.tsx

```diff
@@ -71,6 +83,13 @@ const AudioInputProvider: React.FC<Props> = ({
       audioInputsChanged: async (newAudioInputs: MediaDeviceInfo[]) => {
         logger.info('AudioInputProvider - audio inputs updated');
 
+        if (!shouldAutoSelectAudioInputDevice()) {
```

Comment:
Can you check if you get any use-hook dependency warning from the Terminal?

Reply:
I checked and updated our eslint with `eslint-plugin-react-hooks` with hook related rules as suggested here and updated the `AudioInputProvider` according to warnings. I did not see though warning for use-hook specifically for `shouldAutoSelectAudioInputDevice` in local terminal + browser console.

https://legacy.reactjs.org/docs/hooks-rules.html#eslint-plugin